pkg/xds/envoy/listeners/v3/inbound_listener_configurer.go

```diff
@@ -16,7 +17,7 @@ type InboundListenerConfigurer struct {
 
 func (c *InboundListenerConfigurer) Configure(l *envoy_listener.Listener) error {
 	l.Name = c.ListenerName
-	l.ReusePort = c.Protocol == core_xds.SocketAddressProtocolUDP
+	l.EnableReusePort = util_proto.Bool(c.Protocol == core_xds.SocketAddressProtocolUDP)
```

Comment:
This field seems to be added in Envoy 1.20.x
https://www.envoyproxy.io/docs/envoy/latest/version_history/current

we need to keep compatibility with the latest Kuma major (= Envoy 1.18.x)

Reply:
How do we test that?

Reply:
Should be OK to send both fields in this particular case I think.

Reply:
1) Tests - we have only smoke test for backward compatibility. It would be a good idea to have some infra to also run full E2E with old Kuma DP.
2) If need a feature with later version - I don't think we have ever had this situation so far. You have all the info in DataplaneMetadata so you can always do some if/else in Configurers. We had only such situation with deprecations, that we were keeping deprecations to keep backwards compatibility.
Also like you said, in this case, it's probably fine to have both.